cirq-core/cirq/qis/states.py

```diff
@@ -46,14 +47,70 @@
     # density matrix
     np.ndarray,
     # quantum state object
-    'cirq.QuantumState',
+    'HasQuantumState',
 ]
 document(QUANTUM_STATE_LIKE, """An object representing a quantum state.""")  # type: ignore
 
 
-class QuantumState:
-    """A quantum state.
+class HasQuantumState(abc.ABC):
```

Comment:
Mixed feelings on this name - I'd expect "Has..." to be a method, intuitively. Is `AbstractQuantumState` already taken?